snap/snapcraft.yaml

```diff
@@ -0,0 +1,35 @@
+name: lazygit
+title: Lazygit
+base: core24
+adopt-info: lazygit
+summary: simple terminal UI for git commands
+description: A simple terminal UI for git commands, written in Go with the gocui library.
+license: MIT
+grade: stable
+confinement: classic
+compression: lzo
+contact: 
+  - https://discord.gg/ehwFt2t4wt
+  - https://twitter.com/DuffieldJesse
+donation: https://github.com/sponsors/jesseduffield
+issues: https://github.com/jesseduffield/lazygit/issues
+source-code: https://github.com/jesseduffield/lazygit
+website: https://jesseduffield.com
+
+parts:
+  lazygit:
+    plugin: go
+    source: https://github.com/jesseduffield/lazygit.git
+    source-tag: 'v$SNAPCRAFT_PROJECT_VERSION'
+    source-depth: 1
+    build-snaps:
+      - go
+    override-pull: |
+      craftctl default
+      craftctl set version=$(git describe --tags --abbrev=0 | cut -c 2-)
```

Comment:
This feels a little broad. Can we ensure it actually has the shape vX.Y

Reply:
This will always pull the latest tag. So, you want that if for some reason, you create a tag 45.0 instead of v45.0, the version is still shown as 45.0 right?

Reply:
My concern is that sometimes we will push random tags that aren't actually intended to be releases e.g. pre-releases or things I'm personally testing. So we need to get the latest tag with the correct shape (vX.Y)

Reply:
That's why I think the current implementation is the best. Where you just update the top-level version for the snap and the rest will be taken care of automatically.

Reply:
I'm not following: what's wrong with the override-pull approach that just gets the latest version tag?